blog/src/__tests__/Featured.test.js

```diff
@@ -0,0 +1,56 @@
+import { render, screen} from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import Featured from '../components/Featured/Featured';
+
+const isProperImageURL = (url) => {
+    const regex = /^https?:\/\/[^\s]+\.(jpg|jpeg|png|gif|webp|avif)$/;
+    return regex.test(url);
+  };
+
+describe("Unit tests for Featured component",()=>{
+  
+  it('Dom test for rendering Featured', () => {
+    render(<MemoryRouter><Featured/></MemoryRouter>);
+    const featureElement = screen.getByRole('featured');
+    expect(featureElement).toBeInTheDocument();
+  });
+
+  it('snapshot test to check Featured component', () => {
+    const { asFragment } = render(<MemoryRouter><Featured/></MemoryRouter>);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('Image renders and src is correct', () => {
+    render(<MemoryRouter><Featured/></MemoryRouter>);
+    const image = screen.getByTestId('FeaturedImg');
+    expect(image).toBeInTheDocument();
+    const imageSrc = image.src;
+    expect(isProperImageURL(imageSrc)).toBe(true);
+  });
+
+  it('Image Alternative (alt) is correct', () => {
+    render(<MemoryRouter><Featured/></MemoryRouter>);
+    const Featuredimage = screen.getByTestId('FeaturedImg');
+    expect(Featuredimage).toHaveAttribute('alt', 'Featured image')
+  });
+
+  it('Heading is rendered', () => {
+    render(<MemoryRouter><Featured/></MemoryRouter>);
+    const heading = screen.getByTestId('headingfeatured');
+    expect(heading).toBeInTheDocument();
+  });
+
+  it('Summary is limited to a 250 characters', () => {
+    render(<MemoryRouter><Featured/></MemoryRouter>);
+    const summary = screen.getByTestId('featureSummary');
+    expect(summary.textContent.length).toBeLessThanOrEqual(250);
+  });
```

Comment:
It would be better to create unit tests for CardWide separately. In those tests, you can pass different values for the blog to test all the scenarios. You can pass the content greater than 250 characters equal to 250 characters less than 250 characters and check the behavior of CardWide. Like it should display read more in certain scenarios and shouldn't in some scenraios.